control-plane/mbus-api/src/v0.rs

```diff
@@ -665,6 +665,83 @@ impl From<i32> for Protocol {
         }
     }
 }
+impl From<ReplicaShareProtocol> for Protocol {
+    fn from(src: ReplicaShareProtocol) -> Self {
+        match src {
+            ReplicaShareProtocol::Nvmf => Self::Nvmf,
+        }
+    }
+}
+
+/// The protocol used to share the nexus.
+#[derive(
+    Serialize,
+    Deserialize,
+    Debug,
+    Clone,
+    EnumString,
+    ToString,
+    Eq,
+    PartialEq,
+    Apiv2Schema,
+)]
+#[strum(serialize_all = "camelCase")]
+#[serde(rename_all = "camelCase")]
+pub enum NexusShareProtocol {
+    /// shared as NVMe-oF TCP
+    Nvmf = 1,
+    /// shared as iSCSI
+    Iscsi = 2,
+}
+
+impl Default for NexusShareProtocol {
+    fn default() -> Self {
+        Self::Nvmf
+    }
+}
+impl From<i32> for NexusShareProtocol {
+    fn from(src: i32) -> Self {
+        match src {
+            1 => Self::Nvmf,
+            2 => Self::Iscsi,
+            _ => panic!("Invalid"),
```

Comment:
maybe display what the invalid src was?
same for below

Reply:
Good suggestion. Done.